cranelift/assembler-x64/src/fuzz.rs

```diff
@@ -195,12 +196,16 @@ fn replace_signed_immediates(dis: &str) -> std::borrow::Cow<str> {
             let (_, rest) = chomp("0x", rest); // Skip the '0x' if it's there.
             let n = rest.chars().take_while(char::is_ascii_hexdigit).count();
             let (hex, rest) = rest.split_at(n); // Split at next non-hex character.
-            let simm = match hex.len() {
-                1 | 2 => hex_print_signed_imm!(hex, u8 => i8),
-                4 => hex_print_signed_imm!(hex, u16 => i16),
-                8 => hex_print_signed_imm!(hex, u32 => i32),
-                16 => hex_print_signed_imm!(hex, u64 => i64),
-                _ => panic!("unexpected length for hex: {hex}"),
+            let simm = if dis.starts_with("mov") {
+                u64::from_str_radix(hex, 16).unwrap().to_string()
+            } else {
```

Comment:
Here's another Capstone-related weirdness: the assembler emits all immediates the same but this special logic "fixes" things up so we can match against Capstone. I will be very thankful when we can switch to a more regular disassembler.